src/practices/Rust/RustGitignoreCorrectlySetPractice.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { RustGitignoreCorrectlySetPractice } from './RustGitignoreCorrectlySetPractice';
+import { PracticeEvaluationResult } from '../../model';
+import { TestContainerContext, createTestContainer } from '../../inversify.config';
+
+const GITIGNORE_MISSING = `
```

Comment:
It's not necessary. To check badly set `.gitignore` you can do just the `'...'`

Reply:
I'm actually also checking that it preserves the old value [here](https://github.com/DXHeroes/dx-scanner/blob/b6915355dc58050967b1dba9e7beeb4ef75b6a3f/src/practices/Rust/RustGitignoreCorrectlySetPractice.spec.ts#L79) while writing the new one.

I do this because I found out that when the component detector detects more than one component at the same path (which happens when there is a cargo package that compiles to both an executable and a library at the same path) then the fix handler runs for both components, so it runs twice. Since it runs asynchronously it actually has a **race condition** and appends the fix twice.

Thus I was forced to implement the fix using `writeFile` and overwrite the whole gitignore file so I don't append to it twice. Note that it still actually has a race condition and the only real solution would be to implement some kind of a locking mechanism for fixers to synchronize over.